### PR TITLE
Add platform checks before nested pip call

### PR DESCRIPTION
**Pull request by ddelange**
More verbose errors for #2933 

based on https://github.com/NVIDIA/TensorRT/pull/3080/commits/5334f0dc8b970fd6ec51f75f8fd93c156943d219

### CODE REVIEW

**Comment by ddelange** [comment]
hi @ttyio 👋

is this one also of interest for you?

File: python/packaging/frontend_sdist/setup.py [hunk 1]
```diff
@@ -38,6 +39,22 @@ def run_pip_command(args, call_func):
     except subprocess.CalledProcessError:
         return call_func([os.path.join(sys.exec_prefix, "bin", "pip")] + args)
 
+# cherry-pick information from `packaging.markers.default_environment()` needed to find the right wheel
+# https://github.com/pypa/packaging/blob/23.1/src/packaging/markers.py#L175-L190
+if sys.platform == "linux":
+    platform_marker = "manylinux_2_17"
```
Comment (pranavm-nvidia) [inline python/packaging/frontend_sdist/setup.py:45]: Doesn't seem like these are actually being used anywhere

Comment (ddelange) [reply python/packaging/frontend_sdist/setup.py:45]: yeah they're unused variables (see commit linked), I can clean them up if you like the idea of this PR

**Review by pranavm-nvidia — APPROVED** [review]

**Comment by ddelange** [comment]
^ signed off 👍

**Comment by ddelange** [comment]
Hi @ttyio @zeroepoch :wave:

anything else from my side?

**Comment by zeroepoch** [comment]
@ddelange, We haven't merged this change yet because we have a few tweaks required internally for an upcoming release. The approach looks good though to me. I like that it will better inform users, such as MacOS users that these wheels won't work.

File: python/packaging/frontend_sdist/setup.py [hunk 2]
```diff
@@ -39,6 +40,15 @@ def run_pip_command(args, call_func):
         return call_func([os.path.join(sys.exec_prefix, "bin", "pip")] + args)
 
 
+# check wheel availability using information from https://github.com/pypa/packaging/blob/23.1/src/packaging/markers.py#L175-L190
+if sys.platform != "linux":
+    raise RuntimeError("TensorRT currently only builds wheels for linux")
+if sys.implementation.name != "cpython":
+    raise RuntimeError("TensorRT currently only builds wheels for CPython")
+if platform.machine() != "x86_64":
+    raise RuntimeError("TensorRT currently only builds wheels for x86_64 processors")
```
Comment (ddelange) [inline python/packaging/frontend_sdist/setup.py:49]:
```suggestion
# check wheel availability using information from         
https://github.com/pypa/packaging/blob/23.1/src/packaging/markers.py#L175-L190
if sys.platform not in ("linux", "win32"):
    raise RuntimeError("TensorRT currently only builds wheels for Linux and Windows")
if sys.implementation.name != "cpython":
    raise RuntimeError("TensorRT currently only builds wheels for CPython")
if platform.machine() not in ("x86_64", "AMD64"):
    raise RuntimeError("TensorRT currently only builds wheels for x86_64 processors")
```

File: python/packaging/frontend_sdist/setup.py [hunk 3]
```diff
@@ -39,6 +40,15 @@ def run_pip_command(args, call_func):
         return call_func([os.path.join(sys.exec_prefix, "bin", "pip")] + args)
 
 
+# check wheel availability using information from https://github.com/pypa/packaging/blob/23.1/src/packaging/markers.py#L175-L190
```
[the remark this answers is not in the document]
Comment (ddelange) [reply python/packaging/frontend_sdist/setup.py]: done!

File: python/packaging/frontend_sdist/setup.py [hunk 4]
```diff
@@ -39,6 +40,16 @@ def run_pip_command(args, call_func):
         return call_func([os.path.join(sys.exec_prefix, "bin", "pip")] + args)
 
 
+# check wheel availability using information from         
+https://github.com/pypa/packaging/blob/23.1/src/packaging/markers.py#L175-L190
```
Comment (ttyio) [inline python/packaging/frontend_sdist/setup.py:44]: merge comment line 43 and line 44 to single single, thanks

**Comment by ddelange** [comment]
done!

**Comment by ttyio** [comment]
Thank you @ddelange !

**Pull request merged**